src/main/java/hudson/plugins/warnings/parser/PyLintParser.java

```diff
@@ -34,12 +43,22 @@ protected boolean isLineInteresting(final String line) {
 
     @Override
     protected Warning createWarning(final Matcher matcher) {
-        String message = matcher.group(4);
-        String category = classifyIfEmpty(matcher.group(3), message);
-        //First letter of the Pylint classification is one of F/E/W/R/C. E/F/W are high priority.
-        Priority priority = Priority.NORMAL;
-        if (category.charAt(0) == 'E' || category.charAt(0) == 'F' || category.charAt(0) == 'W') {
-            priority = Priority.HIGH;
+        String message = matcher.group(5);
+        String category = classifyIfEmpty(matcher.group(4), message);
+        Priority priority = null;
+        if (matcher.group(3) != null) {
+            try {
+                priority = Priority.fromString(matcher.group(3));
+            } catch (IllegalArgumentException e) {
+                // will fall back to E/F/W = high
```

Comment:
Of course.  I just added two test cases.  Let me know if you think this is
sufficient.

On Sat, Jun 6, 2015 at 2:13 AM, Ulli Hafner <notifications@github.com>
wrote:

> In src/main/java/hudson/plugins/warnings/parser/PyLintParser.java
> <https://github.com/jenkinsci/warnings-plugin/pull/61#discussion_r31866322>
> :
>
> > @@ -34,12 +43,22 @@ protected boolean isLineInteresting(final String line) {
> >
> >      @Override
> >      protected Warning createWarning(final Matcher matcher) {
> > -        String message = matcher.group(4);
> > -        String category = classifyIfEmpty(matcher.group(3), message);
> > -        //First letter of the Pylint classification is one of F/E/W/R/C. E/F/W are high priority.
> > -        Priority priority = Priority.NORMAL;
> > -        if (category.charAt(0) == 'E' || category.charAt(0) == 'F' || category.charAt(0) == 'W') {
> > -            priority = Priority.HIGH;
> > +        String message = matcher.group(5);
> > +        String category = classifyIfEmpty(matcher.group(4), message);
> > +        Priority priority = null;
> > +        if (matcher.group(3) != null) {
> > +            try {
> > +                priority = Priority.fromString(matcher.group(3));
> > +            } catch (IllegalArgumentException e) {
> > +                // will fall back to E/F/W = high
>
> Can you please add a test case that contains an invalid text as priority.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/warnings-plugin/pull/61/files#r31866322>.
>